File: UIKit and Animations/WWDC 2014/Building Adaptive Apps with UIKit/AdaptivePhotos An Adaptive Application/AdaptivePhotosAnAdaptiveApplication/AdaptivePhotos/AAPLConversation.m
/*
 Copyright (C) 2014 Apple Inc. All Rights Reserved.
 See LICENSE.txt for this sample’s licensing information
 
 Abstract:
 
  The model object that represents a conversation.
  
 */

#import "AAPLConversation.h"
#import "AAPLPhoto.h"

@implementation AAPLConversation

+ (instancetype)conversationWithDictionary:(NSDictionary *)dictionary
{
    AAPLConversation *conversation = [[self alloc] init];
    conversation.name = [dictionary objectForKey:@"name"];
    
    NSArray *photoValues = [dictionary objectForKey:@"photos"];
    NSMutableArray *photos = [NSMutableArray array];
    
    for (NSDictionary *photoValue in photoValues) {
        AAPLPhoto *photo = [AAPLPhoto photoWithDictionary:photoValue];
        [photos addObject:photo];
    }
    
    conversation.photos = photos;
    return conversation;
}

@end

File: UIKit and Animations/WWDC 2014/Building Adaptive Apps with UIKit/AdaptivePhotos An Adaptive Application/AdaptivePhotos_Swift/AdaptivePhotos_Swift/Model/AAPLConversation.swift
/*
Copyright (C) 2014 Apple Inc. All Rights Reserved.
See LICENSE.txt for this sample’s licensing information

Abstract:

The model object that represents a conversation.

*/

import Foundation

class AAPLConversation: NSObject {
	
	var name: String!
	var photos: [AAPLPhoto]!
	
	static func conversationWithDictionary(_ dictionary: [String : Any]) -> AAPLConversation {
		
		let conversation = AAPLConversation()
		conversation.name = dictionary["name"] as! String
		let photoValues = dictionary["photos"] as! [[String : Any]]
		var photos: [AAPLPhoto] = []
		
		for photoValue in photoValues {
			let photo = AAPLPhoto.photoWithDictionary(photoValue)
			photos.append(photo)
		}
		
		conversation.photos = photos
		return conversation
	}
}
